packages/js/src/ws/socket.ts

```diff
@@ -188,10 +189,33 @@ export class Socket extends BaseModule {
 
     this.callWithSession(async () => {
       this.#initializeSocket().catch((error) => {
+        // eslint-disable-next-line no-console
         console.error(error);
       });
 
       return {};
+    }).catch((error) => {
+      // eslint-disable-next-line no-console
+      console.error(error);
+    });
+  }
+
+  disconnect(): void {
+    if (this.#socketIo) {
+      this.#socketIo?.disconnect();
+      this.#socketIo = undefined;
+
+      return;
+    }
+
+    this.callWithSession(async () => {
+      this.#socketIo?.disconnect();
+      this.#socketIo = undefined;
+
+      return {};
+    }).catch((error) => {
+      // eslint-disable-next-line no-console
```

Comment:
otherwise wait for the session to be established and then disconnect

Reply:
Ditto about errors. I don't think we should be handling them.